### PR TITLE
json: fix unquote bug and add test cases.

### PR DESCRIPTION
**Pull request by hicqu**

### CODE REVIEW

File: util/types/json/functions.go [hunk 1]
```diff
@@ -124,6 +129,23 @@ func unquoteString(s string) (string, error) {
 	return ret.String(), nil
 }
 
+// decodeEscapedUnicode decodes unicode into utf8 bytes. e.g.
+// decodeEscapedUnicode("4321") -> "䌡"
+// decodeEscapedUnicode("0034") -> "4"
+func decodeEscapedUnicode(s []byte) (char []byte, err error) {
+	var buffer [6]byte
```
Comment (hanfei1991) [inline util/types/json/functions.go]: Why init len 6 ?

Comment (hicqu) [inline util/types/json/functions.go]: because we want to store a utf8 character, 6 bytes is enough.

Comment (XuHuaiyu) [inline util/types/json/functions.go]: `6` and `2` are hard to understand.
may implement as the demo [here](https://golang.org/pkg/encoding/hex/#Decode) will be easier to understand?

File: util/types/json/functions.go [hunk 2]
```diff
@@ -124,6 +129,23 @@ func unquoteString(s string) (string, error) {
 	return ret.String(), nil
 }
 
+// decodeEscapedUnicode decodes unicode into utf8 bytes. e.g.
+// decodeEscapedUnicode("4321") -> "䌡"
+// decodeEscapedUnicode("0034") -> "4"
+func decodeEscapedUnicode(s []byte) (char []byte, err error) {
```
Comment (hhkbp2) [inline util/types/json/functions.go]: Better to add a test case for this function.

**Comment by hicqu** [comment]
@hanfei1991 @hhkbp2 , PTAL, thanks.

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**